### DISCUSSION TITLE
[Miscellaneous] We have a dead wallet that can't send any tx

### DISCUSSION THREAD

**Discussion by CheshireCatNick** [ethereum-optimism/developers#179] [Miscellaneous]
### Did you check the documentation?

- [x] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

We have a wallet (`0xA42Cd9Ea9830170d98F6cCf68cccCF85D21eBD88`) that is used as a keeper bot. Last week, it sent a few tx when the base fee is high.
And now, we can't send any transaction from this wallet. Even a simple tx to send 0 ETH.

Here's the error message from the provider:
```
cast send --value 0 --rpc-url https://mainnet.optimism.io --nonce 11064 --gas-price 800000000 --priority-gas-price 500000000 --private-key=$PK 0xA42Cd9Ea9830170d98F6cCf68cccCF85D21eBD88
Error:
(code: -32000, message: insufficient funds for gas * price + value: balance 694629222129421825, queued cost 31765732822450692805, tx bumped -150019139003665270, overshot 30921084461317605710, data: None)
```

Based on the error message, it seems like there are some pending transactions from the wallet that is still unconfirmed in the pool, which require around 31.76 ETH gas fee. Is there any way we can overwrite those pending transactions and rescue the fund in this wallet?

**Comment by opfocus** [comment]
I've seen someone report a similar issue at https://discord.com/channels/667044843901681675/1188084808547586119/1188390410180120586. I speculate that it might be due to an unresolved pending transaction. The initial inquirer did not specify whether the issue was eventually resolved

**Reply by CheshireCatNick** [reply]
Agreed. I think the issue is that sum of gas of all pending orders in the pool is ~32 ETH, and we don't have 32 ETH in the wallet. So no matter what gas fee we set to send 0 ETH, it will be rejected by the provider because it checks something like "wallet balance must >= current tx gas + all pending tx gas".
One potential solution might be transferring 32 ETH into this wallet first, and send 0 ETH to clear all pending tx. But I hope there's another solution.

**Comment by sbvegan** [comment]
Hey, we're having the client engineers take a look at this. Do you have a transaction hash from any pending transactions you can share?

**Reply by CheshireCatNick** [reply]
Unfortunately, it's quite hard to find out those pending tx hash on our end. However I just found out some of the pending tx appeared on-chain 1 days ago. They were sent 6 days ago and has stuck in the pool since then.

https://optimistic.etherscan.io/address/0xA42Cd9Ea9830170d98F6cCf68cccCF85D21eBD88

```
cast send --value 0 --rpc-url https://mainnet.optimism.io --nonce 11088 --gas-price 800000000 --priority-gas-price 500000000 --private-key $PK --gas-limit 21000 0xA42Cd9Ea9830170d98F6cCf68cccCF85D21eBD88
Error:
(code: -32000, message: insufficient funds for gas * price + value: balance 692380516390310924, queued cost 15300000000000000000, tx bumped -150066117104852944, overshot 14457553366504836132, data: None)
```

 You can see the queued cost became 15.3 ETH instead of 31.7 ETH before.

**Reply by sbvegan** [reply]
@CheshireCatNick I believe I responded to you or someone on your team in Discord. I'm working with our client engineers to dig deeper into this issue. I'll make sure to follow up with more information as I get it.

**Reply by sbvegan** [reply]
Follow up, we're trying a temporary solution today, but we're doing a deeper investigation here.

**Reply by 42bchen** [reply]
FYI, we can successfully send tx today; seems like the pending tx already dropped by sequencer

**Reply by sbvegan** [reply]
Okay cool! It looks like the fix the client engineer tried worked. We'll dig into this and work so this doesn't happen again.
